modules/gdscript/gdscript_parser.cpp

```diff
@@ -3836,18 +3836,31 @@ bool GDScriptParser::tool_annotation(const AnnotationNode *p_annotation, Node *p
 bool GDScriptParser::icon_annotation(const AnnotationNode *p_annotation, Node *p_node) {
 	ERR_FAIL_COND_V_MSG(p_node->type != Node::CLASS, false, R"("@icon" annotation can only be applied to classes.)");
 	ERR_FAIL_COND_V(p_annotation->resolved_arguments.is_empty(), false);
+
 	ClassNode *p_class = static_cast<ClassNode *>(p_node);
+	String path = p_annotation->resolved_arguments[0];
+
 #ifdef DEBUG_ENABLED
 	if (!p_class->icon_path.is_empty()) {
 		push_error(R"("@icon" annotation can only be used once.)", p_annotation);
 		return false;
 	}
-	if (String(p_annotation->resolved_arguments[0]).is_empty()) {
+	if (path.is_empty()) {
 		push_error(R"("@icon" annotation argument must contain the path to the icon.)", p_annotation->arguments[0]);
 		return false;
 	}
 #endif // DEBUG_ENABLED
-	p_class->icon_path = p_annotation->resolved_arguments[0];
+
+	p_class->icon_path = path;
+
+	if (path.is_empty() || path.is_absolute_path()) {
+		p_class->simplified_icon_path = path.simplify_path();
+	} else if (path.is_relative_path()) {
+		p_class->simplified_icon_path = script_path.get_base_dir().path_join(path).simplify_path();
+	} else {
+		p_class->simplified_icon_path = path;
+	}
+
```

Comment:
@dalexeev Shouldn't `simplified_icon_path` be instead a return value of a `get_simplified_icon_path()`, as the simplified path is necessarily derived from a path.

Reply:
Sorry, I didn't understand what you mean, could you rephrase it please?

Reply:
I mean, a function that returns the simplified path
```cpp
String GDScriptParser::get_simplified_icon_path() {
	if (path.is_empty() || path.is_absolute_path()) {
		return path.simplify_path();
	} else if (path.is_relative_path()) {
		return script_path.get_base_dir().path_join(path).simplify_path();
	}
	return path;
}
```

Reply:
This makes sense to me as well, resolving then :)